src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -476,52 +481,38 @@ internal bool RemoveWorkingDirectoryTrailingCharacter
         /// <param name="settingFileArgIndex">
         /// The index in args to the argument following '-SettingFile'.
         /// </param>
-        /// <param name="parser">
-        /// Used to allow the helper to write errors to the console.  If not supplied, no errors will be written.
-        /// </param>
         /// <returns>
         /// Returns true if the argument was parsed successfully and false if not.
         /// </returns>
-        private static bool TryParseSettingFileHelper(string[] args, int settingFileArgIndex, CommandLineParameterParser parser)
+        private bool TryParseSettingFileHelper(string[] args, int settingFileArgIndex)
         {
             if (settingFileArgIndex >= args.Length)
             {
-                if (parser != null)
-                {
-                    parser.WriteCommandLineError(
-                        CommandLineParameterParserStrings.MissingSettingsFileArgument);
-                }
-
+                SetCommandLineError(CommandLineParameterParserStrings.MissingSettingsFileArgument);
                 return false;
             }
 
-            string configFile = null;
+            string configFile;
             try
             {
                 configFile = NormalizeFilePath(args[settingFileArgIndex]);
             }
             catch (Exception ex)
             {
-                if (parser != null)
-                {
-                    string error = string.Format(CultureInfo.CurrentCulture, CommandLineParameterParserStrings.InvalidSettingsFileArgument, args[settingFileArgIndex], ex.Message);
-                    parser.WriteCommandLineError(error);
-                }
-
+                string error = string.Format(CultureInfo.CurrentCulture, CommandLineParameterParserStrings.InvalidSettingsFileArgument, args[settingFileArgIndex], ex.Message);
+                SetCommandLineError(error);
                 return false;
             }
 
             if (!System.IO.File.Exists(configFile))
             {
-                if (parser != null)
-                {
-                    string error = string.Format(CultureInfo.CurrentCulture, CommandLineParameterParserStrings.SettingsFileNotExists, configFile);
-                    parser.WriteCommandLineError(error);
-                }
-
+                string error = string.Format(CultureInfo.CurrentCulture, CommandLineParameterParserStrings.SettingsFileNotExists, configFile);
+                SetCommandLineError(error);
                 return false;
             }
 
+            SettingsFile = configFile;
```

Comment:
I added some internal properties only for xUnit tests.

Reply:
OK, then please add comment to call out the purpose of those properties.

Reply:
Done.

Reply:
For SettingsFile - I have to move `PowerShellConfig.Instance.SetSystemConfigFilePath();` out of the parser because new tests conflict with test from `test_PSConfiguration.cs` and use SettingsFile.

Reply:
Then you should instead have a test hook, so you will do something like this:

```
if (testhook)
{
    SettingsFile = configFile;
}
else
{
    // call SetSystemConfigFilePath 
}
```